Update README.md

creviera opened the pull request:
## Status

Closes https://github.com/freedomofpress/securedrop-builder/issues/304

## Description of Changes

* Updated references to the old `securedrop-builder` name (I can't even remember it, it's so old ;))
* Since the workstation release management docs were difficult to read and scattered, I did some consolidation work to move release-specific instructions to https://github.com/freedomofpress/securedrop-dev-docs/pull/30, which should be reviewed in tandem with this PR.
* Removed outdated Python build instructions and generalized how to use our build tool

## Checklist

- [ ] Only bookworm tests are failing (see https://github.com/freedomofpress/securedrop-builder/pull/413#issuecomment-1430418078)
- [ ] https://github.com/freedomofpress/securedrop-dev-docs/pull/30 has been merged
- [ ] Link in the readme works (https://developers.securedrop.org/en/latest/workstation_release_management.html#how-to-create-the-dispvm-for-building-packages)
- [ ] visual inspection

gonzalo-bulnes commented:
The changes so far are looking great!

creviera commented:
@gonzalo-bulnes - I spent some time separating release management from generic instructions on how to use this tool. So this PR should be reviewed in tandem with an incoming `securedrop-dev-docs` PR (it'll be in draft because I'll need to edit it for formatting and style still).

creviera commented:
TODO: Once https://github.com/freedomofpress/securedrop-dev-docs/pull/30 is merged, link to the section on how to build a dispVM rather than duplicating the instructions here, see https://github.com/freedomofpress/securedrop-dev-docs/pull/30/files#r1035458154

creviera commented:
[Blocked on https://github.com/freedomofpress/infrastructure/pull/4068 + tandem PR for securedrop repos + actually renaming the repos]

creviera commented:
See https://github.com/freedomofpress/securedrop-builder/pull/413#issuecomment-1430418078

creviera commented:
@gonzalo-bulnes, if you end up merging https://github.com/freedomofpress/securedrop-dev-docs/pull/30, this is the next PR to review, which references a section added in https://github.com/freedomofpress/securedrop-dev-docs/pull/30.

gonzalo-bulnes commented on README.md:
```diff
@@ -1,34 +1,113 @@
 
-# SecureDrop Debian Packaging
-
 [![CircleCI](https://circleci.com/gh/freedomofpress/securedrop-builder/tree/main.svg?style=svg)](https://circleci.com/gh/freedomofpress/securedrop-builder/tree/main)
 
```
Is putting the badge above the main header intentional? (I notice it doesn't show up in the preview.)

creviera replied:
:/ I see it still when I "view file": https://github.com/freedomofpress/securedrop-builder/blob/1f5feba3f9176b17e16084bb7c0877dcd98560bc/README

The badge is intentionally there and this PR shouldn't remove it.

gonzalo-bulnes commented:
Only waiting for https://github.com/freedomofpress/securedrop-dev-docs/pull/30 :slightly_smiling_face:

gonzalo-bulnes commented:
@creviera I see only Bookworm jobs are failing in CI, and FWIW I wouldn't block this PR on them.

creviera commented:
@gonzalo-bulnes, while addressing your reviews, I noticed another opportunity to further consolidate our documentation. For deployment, the readme now points to the docs in https://github.com/freedomofpress/securedrop-dev-docs/pull/30, so you'll notice that last commit that I pushed is mostly just deleting stuff: https://github.com/freedomofpress/securedrop-builder/pull/400/commits/ba2c611bf6dbcd6e96d310027f1bb0e0e42bc280

gonzalo-bulnes reviewed (approved):
- [x] Only bookworm tests are failing (see https://github.com/freedomofpress/securedrop-builder/pull/413#issuecomment-1430418078)
- [x] https://github.com/freedomofpress/securedrop-dev-docs/pull/30 has been merged
- [x] Link in the readme works (https://developers.securedrop.org/en/latest/workstation_release_management.html#how-to-create-the-dispvm-for-building-packages)
- [x] visual inspection
